abcEconomics/trade.py

```diff
@@ -40,19 +40,23 @@
 # ***************************************************************************************** #
 import random
 from collections import defaultdict, OrderedDict
+
+import cython
+
 from abcEconomics.notenoughgoods import NotEnoughGoods
 
-epsilon = 0.00000000001
+epsilon: cython.double = 0.00000000001
 
 
 def get_epsilon():
     return epsilon
 
 
+# @cython.cclass
 class Offer(object):
-    __slots__ = ('sender', 'receiver', 'good', 'quantity', 'price', 'currency',
```

Comment:
Why net keeping slots for pypy/non cython compilation?

Reply:
It might slow down the cython version.